From: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Date: Sun, 1 Nov 2020 20:50:14 +0100
Subject: Change program path for udevadm from /usr/bin to /bin

---
 scripts/ofpathname | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/ofpathname b/scripts/ofpathname
index 2ceae25..077de3f 100755
--- a/scripts/ofpathname
+++ b/scripts/ofpathname
@@ -330,7 +330,7 @@ is_pata_dev()
     local sysfs_dir
     local udev_path
     local udevinfo="/usr/bin/udevinfo"
-    local udevadm="/usr/bin/udevadm"
+    local udevadm="/bin/udevadm"
 
     if [[ -a $udevadm ]]; then
         udev_path=`$udevadm info --query=path --name=$DEVNAME`
